wrappers/csharp/cs-tutorial-8-D400-on-chip-calibration/Window.xaml.cs

```diff
@@ -0,0 +1,178 @@
+
+                    SetupWindow(pp, out updateDepth, out updateColor);
+                }
+
+                // Rendering task
+                var renderingPause = false;
+                var rendering = Task.Factory.StartNew(() =>
+                {
+                    while (!tokenSource.Token.IsCancellationRequested)
+                    {
+                        if (renderingPause) continue; //pause the rendering
+
+                        // We wait for the next available FrameSet and using it as a releaser object that would track
+                        // all newly allocated .NET frames, and ensure deterministic finalization
+                        // at the end of scope. 
+                        using (var frames = pipeline.WaitForFrames())
+                        {
+                            var colorFrame = frames.ColorFrame.DisposeWith(frames);
+                            var depthFrame = frames.DepthFrame.DisposeWith(frames);
+
+                            // We colorize the depth frame for visualization purposes
+                            var colorizedDepth = colorizer.Process<VideoFrame>(depthFrame).DisposeWith(frames);
+
+                            // Render the frames.
+                            if (colorFrame != null)
+                                Dispatcher.Invoke(DispatcherPriority.Render, updateDepth, colorizedDepth);
+                            if (depthFrame != null)
+                                Dispatcher.Invoke(DispatcherPriority.Render, updateColor, colorFrame);
+
+                            Dispatcher.Invoke(new Action(() =>
+                            {
+                                String depth_dev_sn = depthFrame.Sensor.Info[CameraInfo.SerialNumber];
+                                txtTimeStamp.Text = $"{depth_dev_sn} : {depthFrame.Timestamp,-20:0.00}({depthFrame.TimestampDomain})" +
+                                $"{Environment.NewLine}To start Auto-Calibration flow, switch focus to the application console and press C";
+                            }));
+                        }
+                    }
+                }, tokenSource.Token);
+
+                // Input to calibration mode task
+                Task.Factory.StartNew(() =>
+                {
+                    while (!tokenSource.Token.IsCancellationRequested)
+                    {
+                        if (ConsoleKey.C == ExampleAutocalibrateDevice.ConsoleGetKey(new[] { ConsoleKey.C},
+                                         "To start Auto-Calibration flow, switch focus to the application console and press C"))
+                        {
+                            renderingPause = true;
+                            Console.WriteLine($"{Environment.NewLine}Stopping rendering pipeline...");
+                            pipeline.Stop();
+
+                            new ExampleAutocalibrateDevice().Start();
+
+                            Console.WriteLine($"{Environment.NewLine}Starting rendering pipeline...");
```

Comment:
Consider changing the printed text so that it would include the calibration, something like: "Starting rendering pipeline for calibration..."

Reply:
Not essential